### PR TITLE
Unr 3350 move settings from ue to plugin

### PR DESCRIPTION
**Pull request by yunjielu**
#### Description
Move UnrealEngine SpatialNetworking to Plugin SpatialGDKSetting

Engine PR: https://github.com/improbableio/UnrealEngine/pull/470

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKServices/SpatialGDKServices.Build.cs [hunk 1]
```diff
@@ -29,7 +31,8 @@ public SpatialGDKServices(ReadOnlyTargetRules Target) : base(Target)
 				"Json",
 				"JsonUtilities",
 				"UnrealEd",
-				"Sockets"
+				"Sockets",
+                "SpatialGDK"
```
Comment (raymonwang) [inline SpatialGDK/Source/SpatialGDKServices/SpatialGDKServices.Build.cs:35]: Format the code.

File: SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h [hunk 2]
```diff
@@ -344,6 +344,9 @@ class SPATIALGDK_API USpatialGDKSettings : public UObject
 	UPROPERTY(Config)
 	bool bUseSpatialView;
 
+	UPROPERTY(EditAnywhere, config, Category = "Connection")
+	bool bSpatialNetworking;
```
Comment (raymonwang) [inline SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h:348]: Set a default value to this parameter if Unreal doesn't load a value from INI file.

Comment (yunjielu) [reply SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h:348]: Assign the default true value in the initializing list

File: SpatialGDK/Source/SpatialGDKServices/SpatialGDKServices.Build.cs (hunk 1 above)
Comment (yunjielu) [reply SpatialGDK/Source/SpatialGDKServices/SpatialGDKServices.Build.cs:35]: I revert this cs file as can not directly access SpatialGDK module in SpatialGDKServices, will result in "failed to load SpatialGDKService" issue when starting editor, looks like a circular dependent problem, instead using module manager to access SpatialGDK module in SpatialGDKService

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp [hunk 3]
```diff
@@ -99,7 +100,7 @@ void FSpatialGDKEditorToolbarModule::StartupModule()
 
 	FEditorDelegates::PreBeginPIE.AddLambda([this](bool bIsSimulatingInEditor)
 	{
-		if (GIsAutomationTesting && GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking())
+		if (GIsAutomationTesting && USpatialStatics::IsSpatialNetworkingEnabled())
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp:103]: The `SpatialStatics` version only exists for blueprint access, we can continue using the `GetDefault<USpatialGDKSettings>()->bSpatialNetworking;` style and avoid having an additional include.

File: SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp [hunk 4]
```diff
@@ -318,14 +320,16 @@ class SchemaTestFixture
 	void EnableSpatialNetworking()
 	{
 		UGeneralProjectSettings* GeneralProjectSettings = GetMutableDefault<UGeneralProjectSettings>();
-		bCachedSpatialNetworking = GeneralProjectSettings->UsesSpatialNetworking();
-		GeneralProjectSettings->SetUsesSpatialNetworking(true);
+		bCachedSpatialNetworking = USpatialStatics::IsSpatialNetworkingEnabled();
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp:323]: Same here and in the rest of the PR, I prefer the previous usage of `GetDefault<USpatialGDKSettings>()->bSpatialNetworking;`

**Comment by joshuahuburn** [comment]
This is quite a substantial change, what's the benefit?

File: SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp [hunk 5]
```diff
@@ -77,7 +77,7 @@ void FLocalDeploymentManager::Init(FString RuntimeIPToExpose)
 			TryStopSpatialService();
 			// Start spatial service in the current project if spatial networking is enabled
 
-			if (GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking())
+			if (SpatialEngineHelper::UsesSpatialNetworking())
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp:80]: This seems a bit strange. If this PR wants to move the `bSpatialNetworking` flag why not put it in the `SpatialGDKServicesSettings` module which can be used by all other modules (including this one).

Comment (joshuahuburn) [reply SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp:80]: Also how is `SpatialEngineHelper` being imported?

Comment (joshuahuburn) [reply SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp:80]: Ignore the `SpatialGDKServicesSetting` comment

**Pull request closed without merging**